inc/class-core-author-block.php

```diff
@@ -0,0 +1,277 @@
+<?php
+/**
+ * Core Author Block class file
+ *
+ * This class filters the core post author block so that it displays info from the Byline Manager.
+ * It also duplicates the core post author block as many times as necessary to display all the bylines.
+ *
+ * @package Byline_Manager
+ */
+
+declare( strict_types = 1 );
+
+namespace Byline_Manager;
+
+/**
+ * Sets up the Singleton trait use.
+ *
+ * @phpstan-use-trait-for-class Singleton
+ */
+use Byline_Manager\traits\Singleton;
+use WP_Post;
+use DOMDocument;
+use DOMNodeList;
+use DOMXPath;
+
+/**
+ * Conditions
+ */
+class Core_Author_Block {
+	/**
+	 * Use the singleton.
+	 *
+	 * @use Singleton<static>
+	 */
+	use Singleton;
+
+	/**
+	 * Array of bylines.
+	 *
+	 * @var array
+	 */
+	public static array $bylines = [];
+
+	/**
+	 * Boolean that controls when to render the author blocks.
+	 *
+	 * @var bool
+	 */
+	public static bool $render = true;
+
+	/**
+	 * Initializes
+	 *
+	 * Gets required bylines post meta and setups a filter.
+	 *
+	 * @return void
+	 */
+	public function init(): void {
+		global $post;
+
+		if ( ! empty( $post ) && property_exists( $post, 'ID' ) ) {
+			// Get the byline meta and set it to our property.
+			$this::$bylines = get_post_meta( $post->ID, 'byline', true );
+		}
+
+		add_filter( 'render_block_core/post-author', [ $this, 'append_and_filter_post_author_blocks' ], 10, 2 );
+	}
+
+	/**
+	 * Filters the post author block data to use the bylines, also duplicates the post author block as many times
+	 * as necessary to output all the bylines.
+	 *
+	 * @param string               $block_content The block content.
+	 * @param array<string, mixed> $block The full block, including name and attributes.
+	 *
+	 * @return string The block content.
+	 */
+	public function append_and_filter_post_author_blocks( string $block_content, array $block ): string {
+		// Check that render is true and that we have bylines.
+		if ( $this::$render && ! empty( $this::$bylines['profiles'] ) ) {
+			// Count the bylines.
+			$bylines_count = count( $this::$bylines['profiles'] );
+
+			// A string variable that will be used for the final rendered content.
+			$additional_blocks = '';
+
+			// Loop through the total bylines count.
+			for ( $i = 0; $i < $bylines_count; $i++ ) {
+				// Filter the block and append it to our string variable.
+				$additional_blocks .= $this->filter_post_author_block( $block_content, $i );
+			}
+
+			// Replace the original block with the new blocks.
+			$block_content = $additional_blocks;
+
+			// Update our boolean to not render anymore.
+			$this::$render = false;
+		} else {
+			return '';
+		}
+
+		return $block_content;
```

Comment:
I'd swap this logic. Moving the `block_content` to return inside the conditional.

Reply:
Swapping this essentially breaks this feature. So it's not a suggestion I can implement. Essentially if I have two bylines and two author block, it ends up rendering nothing, if I have two bylines and two author blocks, I renders just one.